src/wallet/test/silentpayment_tests.cpp

```diff
@@ -0,0 +1,161 @@
+            }
+
+            std::vector<std::pair<CKey, bool>> sender_secret_keys;
+            for (const auto& key : given["input_priv_keys"].getValues()) {
+                sender_secret_keys.emplace_back(ParseHexToCKey(key[0].get_str()), key[1].get_bool());
+            }
+            std::vector<CRecipient> silent_payment_addresses;
+            for (const auto& recipient : given["recipients"].getValues()) {
+                std::string silent_payment_address = recipient[0].get_str();
+                CAmount amount = recipient[1].get_real() * COIN;
+                const auto& sp = DecodeDestination(silent_payment_address);
+                silent_payment_addresses.push_back(CRecipient{sp, amount, false});
+            }
+
+            // silent payments logic
+            CKey scalar_ecdh_input = PrepareScalarECDHInput(sender_secret_keys, outpoints);
+
+            std::map<size_t, V0SilentPaymentDestination> sp_dests;
+            for (size_t i = 0; i < silent_payment_addresses.size(); ++i) {
+                if (const auto* sp = std::get_if<V0SilentPaymentDestination>(&silent_payment_addresses.at(i).dest)) {
+                    sp_dests[i] = *sp;
+                }
+            }
+            std::map<size_t, WitnessV1Taproot> sp_tr_dests = GenerateSilentPaymentTaprootDestinations(scalar_ecdh_input, sp_dests);
+
+            for (const auto& [out_idx, tr_dest] : sp_tr_dests) {
+                assert(out_idx < silent_payment_addresses.size());
+                silent_payment_addresses[out_idx].dest = tr_dest;
+            }
+
+            std::vector<CRecipient> expected_spks;
+            for (const auto& recipient : expected["outputs"].getValues()) {
+                std::string pubkey_hex = recipient[0].get_str();
+                CAmount amount = recipient[1].get_real() * COIN;
+                auto tap = WitnessV1Taproot(XOnlyPubKey(ParseHex(pubkey_hex)));
+                expected_spks.push_back(CRecipient{tap, amount, false});
+            }
+
+            BOOST_CHECK(silent_payment_addresses.size() == expected_spks.size());
+            for (const auto& spk : silent_payment_addresses) {
+                BOOST_CHECK(std::find(expected_spks.begin(), expected_spks.end(), spk) != expected_spks.end());
+            }
+        }
+
+        // Test receiving
+        for (const auto& recipient : vec["receiving"].getValues()) {
+            // TODO: implement labels for Bitcoin Core, until then skip the receiving with labels tests
+            if (recipient["supports_labels"].get_bool()) {
+                BOOST_TEST_MESSAGE("Labels not implemented; skipping..");
+                continue;
+            }
+
+            const auto& given = recipient["given"];
+            const auto& expected = recipient["expected"];
+
+            std::vector<COutPoint> outpoints;
+            for (const auto& outpoint : recipient["given"]["outpoints"].getValues()) {
```

Comment:
In 78d1426fabd7b5b6cdd28de508a1692fce291174 "Add BIP352 test vectors as unit tests"

nit: `s/recipient["given"]/given`

```suggestion
            for (const auto& outpoint : given["outpoints"].getValues()) {
````

Reply:
Done